cache-filter/src/filter/root.rs

```diff
@@ -1,18 +1,22 @@
 use crate::configuration::FilterConfig;
 use crate::filter::http::CacheFilter;
 use crate::rand::thread_rng::{thread_rng_init_fallible, ThreadRng};
-use crate::unique_callout::{WaiterAction, WAITING_CONTEXTS};
-use crate::utils::request_process_failure;
 use crate::{debug, info, warn};
 use proxy_wasm::{
-    hostcalls::set_effective_context,
     traits::{Context, HttpContext, RootContext},
     types::{ContextType, LogLevel},
 };
-use threescale::{
-    proxy::{get_app_id_from_cache, get_application_from_cache, CacheKey},
-    stats::*,
-    structs::{AppIdentifier, ThreescaleData},
+use threescale::{proxy::CacheKey, stats::*, structs::ThreescaleData};
+// Following imports are only needed when unqiue-callout feature is enabled.
+#[cfg(feature = "unique_callout")]
+use {
+    crate::unique_callout::{WaiterAction, WAITING_CONTEXTS},
+    crate::utils::request_process_failure,
+    proxy_wasm::hostcalls::set_effective_context,
+    threescale::{
+        proxy::{get_app_id_from_cache, get_application_from_cache},
+        structs::AppIdentifier,
+    },
```

Comment:
If this is only needed for `on_queue_ready`, then we should `use` them only within the body of `on_queue_ready`.

Reply:
Way better!